### DISCUSSION TITLE
trying to configure typescript's tsserver section in :initializationOptions. Doesn't work and can't debug it

### DISCUSSION THREAD

**Discussion by gdanov** [joaotavora/eglot#1457] [Q&A]
I'm trying to make typescript lang server dump a log. Also trying some other hacks and realize I can't reliably know if I configured it right (regardless if via workspace or something else).

here is my (not working as I expect it) config (sorry for the tabs):

```
(add-to-list 'eglot-server-programs
								 ;; WARN this can be overriden in .dir-locals !!
								 '(typescript-mode  "yarn typescript-language-server" "--stdio"
																		:initializationOptions
																		(:tsserver
																		 (
																				;:path "./node_modules/typescript/lib/tsserver.js"
																			:logVerbosity "verbose"
																			:logDirectory "./tsserver.log"
																			)
																		 
																		 :preferences 
																		 (;;https://github.com/typescript-language-server/typescript-language-server#initializationoptions
																			:includeCompletionsForModuleExports  nil
																			:includeCompletionsForImportStatements  t
																			
																			;; experimenting ... but seems to have no effect
																			:includeInlayParameterNameHints:  "all"
																			:includeInlayParameterNameHintsWhenArgumentMatchesName t
																			:includeInlayFunctionParameterTypeHints t
																			:includeInlayVariableTypeHints t
																			:includeInlayVariableTypeHintsWhenTypeMatchesName t
																			:includeInlayPropertyDeclarationTypeHints t
																			:includeInlayFunctionLikeReturnTypeHints t
																			:includeInlayEnumMemberValueHints t

																			;; not sure this is the right section
																			:noErrorTruncation t
																			
																			:codeActionsOnSave (
																													:source.organizeImports  t
																													:source.removeUnusedImports.ts  t)
																			))
																		)
								 )
```

problem is eglot events buffer shows this

```
(:jsonrpc "2.0" :id 1 :method "initialize" :params
					(:processId 9215 :rootPath "/Users/gdanov/work/my-project/backend/" :rootUri "file:///Users/gdanov/work/my-project/backend" :initializationOptions #s(hash-table size 1 test eql rehash-size 1.5 rehash-threshold 0.8125 data ())
```

so I have absolutely no idea what config went to the server. See initializationOptions content — hashtable? What am I doing wrong?

**Comment by manuel-uberti** [comment]
I suggest going through [the manual](https://joaotavora.github.io/eglot/#Reporting-bugs-1) first in order to provide all the information needed to start debugging properly.

**Reply by gdanov** [reply]
I really need to do all that in order to get answer to the question "seeing #s(hash-table size 1 test  ...  in the events log is normal yes/no" ?

**Reply by manuel-uberti** [reply]
Well, I tend to reply with that link to all the reports here that at a cursory look do not seem helpful to @joaotavora (who takes care of most the job anyway).

If @joaotavora is able to help you without all the information in that link, then it's fine with me.

**Reply by joaotavora** [reply]
I can't.  It could be normal, and it could not be.  Indeed I have absolutely no idea what's being attempted here, and I won't even consider spend time looking into it unless it is subsumed into something that I might consider looking into :-)
